lib/cfg-tree.c

```diff
@@ -1082,7 +1083,6 @@ cfg_tree_compile_junction(CfgTree *self,
         }
     }
 
-  *outer_pipe_head = &fork_mpx->super;
```

Comment:
Discussed IRL. We fortunately get away with the NULL pointer dereferation, and set `outer_pipe_head` to NULL in those cases, due to a compiler optimization. (Also checked in assembly.)
Also checked every calling point of `cfg_tree_compile_junction` since now we "lost the initialization" of `outer_pipe_head`.